src/main/java/fruitshop/model/FruitTransaction.java

```diff
@@ -0,0 +1,69 @@
+package fruitshop.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public FruitTransaction() {
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public FruitTransaction setOperation(Operation operation) {
+        this.operation = operation;
+        return this;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public FruitTransaction setFruit(String fruit) {
+        this.fruit = fruit;
+        return this;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public FruitTransaction setQuantity(int quantity) {
+        this.quantity = quantity;
+        return this;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private final String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public static Operation getOperation(String code) {
+            for (Operation operation : Operation.values()) {
+                if (operation.getCode().equals(code)) {
+                    return operation;
+                }
+            }
+            throw new IllegalArgumentException("Unknown code " + code);
+        }
+
+        public String getCode() {
+            return code;
+        }
+    }
+}
```

Comment:
The class `FruitTransaction` should be in a package that groups similar classes. Consider placing it in a package named `model` to maintain a clean project structure.